### DISCUSSION TITLE
C++ and compiler requirements

### DISCUSSION THREAD

**Discussion by Loara** [chaotic-society/theoretica#13] [General]
What are the minimum features required for compilers in order to compile and use this library?

**Comment by mattiaisgro** [comment]
Basic std classes like std::vector and std::array are needed (although I'm planning to make their use optional), inline x86 assembly (MSVC or GCC) is currently needed but I'm implementing fallback implementations which will make it optional too (for architecture independency). There are no other dependencies. Any recent (last few years) version of GCC on a x86 architecture, regardless of system, should compile just fine.

**Comment by mattiaisgro** [comment]
The library now works with at least C++11 support. Some features which depend on C++14 get turned on or off automatically during compilation.

**Comment by mattiaisgro** [comment]
Since we now use the auto keyword in return statements, the minimum required version is C++14.

**Comment by mattiaisgro** [comment]
@Loara following the new [Roadmap for 2026](https://github.com/chaotic-society/theoretica/blob/master/txt%2FROADMAP.md), the library is moving to dual support for C++14 and C++20. Core features will remain C++14 compatible.
